runtime/lib/registry/Registry.cpp

```diff
@@ -72,6 +72,32 @@ class LibraryManager {
     }
 };
 
+inline const uintptr_t ext_macos()
+{
+#ifdef __APPLE__
+    return reinterpret_cast<uintptr_t>(".dylib");
+#else
+    return reinterpret_cast<uintptr_t>(nullptr);
+#endif
+}
+
+inline const uintptr_t ext_unix()
+{
+#ifdef __linux__
+    return reinterpret_cast<uintptr_t>(".so");
+#else
+    return reinterpret_cast<uintptr_t>(nullptr);
+#endif
+}
+
+inline const char *ext()
+{
+    const uintptr_t ext_target = ext_macos() | ext_unix();
+    return reinterpret_cast<char *>(ext_target);
+}
+
+std::string library_name(std::string name) { return name + ext(); }
```

Comment:
Is it not possible to have a single function that chooses between .so and .dylib? Not sure but maybe something like:  with #ifdef and #elif?

Reply:
Could we write something like or is it not working?

inline const char* ext() {
#ifdef __APPLE__
    return ".dylib";
#elif __linux__
    return ".so";
#else
    return nullptr;
#endif
}

Reply:
It works, I never said it didn't. I just like my style better to avoid having a lot of ifdef's. I'll take this as you have a strong preference to do it this way. Fine, let's rewrite that way.

Reply:
I have to agree that the current style is more confusing because the decision is spread across several functions whose output has to processed together, compared to the single decision table proposed by Romain.

Reply:
I don't mind the solution too much it was more for personal knowledge :+1:

Reply:
https://github.com/PennyLaneAI/catalyst/pull/649/commits/00267b6321709c5c1e8e371cd522737119811545